cmd/describe_test.go

```diff
@@ -0,0 +1,142 @@
+package cmd
+
+import (
+	"testing"
+
+	fn "knative.dev/func"
+	"knative.dev/func/mock"
+)
+
+// TestDescribe_ByName ensures that describing a function by name invokes
+// the describer appropriately.
+func TestDescribe_ByName(t *testing.T) {
+	var (
+		testname  = "testname"
+		describer = mock.NewDescriber()
+	)
+
+	describer.DescribeFn = func(n string) (fn.Instance, error) {
+		if n != testname {
+			t.Fatalf("expected describe name '%v', got '%v'", testname, n)
+		}
+		return fn.Instance{}, nil
+	}
+
+	cmd := NewDescribeCmd(NewClientFactory(func() *fn.Client {
+		return fn.New(fn.WithDescriber(describer))
+	}))
+
+	cmd.SetArgs([]string{testname})
+	if err := cmd.Execute(); err != nil {
+		t.Fatal(err)
+	}
+
+	if !describer.DescribeInvoked {
+		t.Fatal("Describer not invoked")
+	}
+}
+
+// TestDescribe_ByProject ensures that describing the currently active project
+// (func created in the current working directory) invokes the describer with
+// its name correctly.
+func TestDescribe_ByProject(t *testing.T) {
+	root := fromTempDirectory(t)
+
+	err := fn.New().Create(fn.Function{
+		Name:     "testname",
+		Runtime:  "go",
+		Registry: TestRegistry,
+		Root:     root,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	describer := mock.NewDescriber()
+	describer.DescribeFn = func(n string) (i fn.Instance, err error) {
+		if n != "testname" {
+			t.Fatalf("expected describer to receive name 'testname', got '%v'", n)
+		}
+		return
+	}
+	cmd := NewDescribeCmd(NewClientFactory(func() *fn.Client {
+		return fn.New(fn.WithDescriber(describer))
+	}))
+	if err := cmd.Execute(); err != nil {
```

Comment:
```suggestion
	}))
	cmd.SetArgs([]string{})
	if err := cmd.Execute(); err != nil {
```

Reply:
Ignoring current user's KUBECONFIG and clearing args from IDEs are now being addressed in separate PRs